onnx/checker.cc

```diff
@@ -35,15 +43,138 @@ namespace checker {
     }                                         \
   } while (0)
 
-void check_value_info(const ValueInfoProto& value_info, const CheckerContext&) {
+bool check_name_syntax(const std::string& name) {
+  if (name.empty()) {
+    return false;
+  }
+
+  bool correct = true;
+
+  // Names should adhere to C identifier syntax.
+  auto iter = name.cbegin();
+
+  char c = *iter;
+  if (!(isalpha(c) || c == '_')) {
+    correct = false;
+  }
+
+  ++iter;
+
+  for (; iter < name.cend(); ++iter) {
+    c = *iter;
+    if (!(isalnum(c) || c == '_')) {
+      correct = false;
+    }
+  }
+
+  return correct;
+}
+
+bool check_domain_syntax(const std::string& name) {
+  if (name.empty()) {
+    return false;
+  }
+
+  bool correct = true;
+
+  // Names should adhere to C identifier syntax.
+  auto iter = name.cbegin();
+
+  char c = *iter;
+  if (!isalpha(c)) {
+    correct = false;
+  }
+
+  ++iter;
+
+  for (; iter < name.cend(); ++iter) {
+    c = *iter;
+    if (!(isalnum(c) || c == '.')) {
+      correct = false;
+    }
+  }
+
+  return correct;
+}
+
+#define enforce_c_identifier(proto, field)       \
+  do {                                           \
+    const auto& str = proto.field();             \
+    if (!check_name_syntax(str)) {               \
```

Comment:
It's weird to call `check_name_syntax` in `enforce_c_identifier`.
We have to two options to solve the problem:
1) change `check_name_syntax` to `check_c_identifier`
2) change `enforce_c_identifier` to `enforace_name_syntax`

Reply:
Excellent point. Going with your second suggestion.